### PR TITLE
Chore: Remove bus from contexthandler

### PR DESCRIPTION
**Pull request by zserge**
Related to #44130

### CODE REVIEW

File: pkg/services/sqlstore/mockstore/mockstore.go [hunk 1]
```diff
@@ -44,7 +44,8 @@ type SQLStoreMock struct {
 	ExpectedUserStars              map[int64]bool
 	ExpectedLoginAttempts          int64
 
-	ExpectedError error
+	ExpectedError            error
+	ExpectedSetUsingOrgError error
```
Comment (sakjur) [inline pkg/services/sqlstore/mockstore/mockstore.go:48]: Curious, why is this change needed?

**Review by sakjur — APPROVED** [review]

**Review by papagian — APPROVED** [review]
LGTM

File: pkg/services/sqlstore/mockstore/mockstore.go (hunk 1 above)
Comment (zserge) [reply pkg/services/sqlstore/mockstore/mockstore.go:48]: One terrible unit tests that expects storage to succeed on all lookups, but fail on SetUsingOrg 🤷 
Better leave this until the store split than figure out how to fix that test.

**Review by spinillos — APPROVED** [review]

File: pkg/services/sqlstore/mockstore/mockstore.go (hunk 1 above)
Comment (zserge) [reply pkg/services/sqlstore/mockstore/mockstore.go:48]: The relationship between OrgRedirect and ContextHandler should be revisited, currently they rely on too many implementation details which leads to fragile tests and complicates the refactoring.

**Comment by sakjur** [comment]
@zserge Should this be backported to 8.5.0 or should the milestone move?
